File: src/test/new_obj.cc
#include "test/new_obj.hh"

#include <iostream>

namespace gem5
{

NewObj::NewObj(const NewObjParams &params) :
    SimObject(params),
    param_1(params.param_1)
{
    std::cout << "A new SimObject! param_1:"<<param_1<< std::endl;
}

}

File: src/test/NewObj.py
from m5.params import *
from m5.SimObject import SimObject

class NewObj(SimObject):
    type = 'NewObj'
    cxx_header = "test/new_obj.hh"
    cxx_class = "gem5::NewObj"
    param_1 = Param.Int(1, "param_1")

